programs/cli_wallet/main.cpp

```diff
@@ -178,9 +178,9 @@ int unsafe_main(int argc, char** argv) {
     for (auto& name_formatter : wapiptr->get_result_formatters())
         wallet_cli->format_result(name_formatter.first, name_formatter.second);
 
-    boost::signals2::scoped_connection closed_connection(con->closed.connect([=]{
-        cerr << "Server has disconnected us.\n";
-        wallet_cli->stop();
+    boost::signals2::scoped_connection closed_connection(con->closed.connect([=,&client]{
+        cerr << "Server has disconnected us, reconnecting.\n";
+        client.connect(wdata.ws_server, true);
```

Comment:
Can we change logic to reconnect on a `new command`, not on `losing connection`?

IMHO, it isn't good strategy to keep live-connection all time.

For example:
1. On losing connection:
1.1 Show error message
1.2 Set flag to `disconnected`
2. On a `new command` if connection is `disconnected` - restore connection.

Reply:
I want to implement the same strategy you described, but it's not (easily) possible now. There is visibility problem, so connecting class can't restore connection itself, it can only use provided connection. I believe this can be fixed with #484/7